libs/wingc/src/jsify.rs

```diff
@@ -1476,13 +1524,25 @@ impl Display for Component<'_> {
 }
 
 impl<'a> FieldReferenceVisitor<'a> {
-	fn analyze_expr(node: &'a Expr) -> Vec<Component> {
+	fn analyze_expr(&self, node: &'a Expr) -> Vec<Component> {
 		match &node.kind {
 			ExprKind::Reference(Reference::Identifier(x)) => {
+				let scope = match &self.function_def.body {
+					FunctionBody::Statements(scope) => scope,
+					FunctionBody::External(_) => panic!("unexpected expression inside body of extern functions"),
+				};
+				let env = scope.env.borrow();
+				let env = env.as_ref().expect("scope should have an env");
```

Comment:
Wow this is interesting... overriding `env`... I guess that's legitimate.

Reply:
Ah yeah, this is a Rust trick. You can also do stuff like this:
```rust
let mut xs = vec![1, 2, 3];
xs.push(4);
let xs = xs; // no longer can be changed!
```
It's pretty neat™️

Reply:
Why not. It's safe